contrib/guix/libexec/codesign.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+export LC_ALL=C
+set -e -o pipefail
+export TZ=UTC
+
+if [ -n "$V" ]; then
+    # Print both unexpanded (-v) and expanded (-x) forms of commands as they are
+    # read from this file.
+    set -vx
+fi
+
+# Check that required environment variables are set
+cat << EOF
+Required environment variables as seen inside the container:
+    HOST: ${HOST:?not set}
+    SOURCE_DATE_EPOCH: ${SOURCE_DATE_EPOCH:?not set}
+    DISTSRC: ${DISTSRC:?not set}
+    OUTDIR: ${OUTDIR:?not set}
+    DETACHED_SIG_REPO_DIR: ${DETACHED_SIG_REPO_DIR:?not set}
+    UNSIGNED_TARBALL: ${UNSIGNED_TARBALL:?not set}
+EOF
+
+UNSIGNED_DIR=/bitcoin/unsigned
+
+# Define DISTNAME variable.
+# shellcheck source=contrib/gitian-descriptors/assign_DISTNAME
+# source "$DISTSRC"/contrib/gitian-descriptors/assign_DISTNAME
+
+# case "$HOST" in
+#     *mingw*)
+#         UNSIGNED_TARBALL="${OUTDIR}/${DISTNAME}-win-unsigned.tar.gz"
+#         ;;
+#     *darwin*)
+#         UNSIGNED_TARBALL="${OUTDIR}/${DISTNAME}-osx-unsigned.tar.gz"
+#         ;;
+# esac
+
+mkdir -p "$UNSIGNED_DIR"
+tar -C "$UNSIGNED_DIR" -xf "$UNSIGNED_TARBALL"
+
+case "$HOST" in
+    *mingw*)
+        SIGDIR="$DETACHED_SIG_REPO_DIR"/win
+
+        find "$UNSIGNED_DIR" -name "*-unsigned.exe" | while read -r infile; do
+            infile_base="$(basename "$infile")"
+            osslsigncode attach-signature -in "$infile" -out "${OUTDIR}/${infile_base/-unsigned}" -sigin "${SIGDIR}/${infile_base}.pem"
+        done
+        ;;
+    *darwin*)
+        export SIGDIR="$DETACHED_SIG_REPO_DIR"/osx
+
+        export SIGNED_APP_DIST_DIR="$SIGDIR"/dist
+        export OUT_SIGNED_DMG="$OUTDIR"/bitcoin-osx-signed.dmg
+
+        cd "$UNSIGNED_DIR"  # URGH
+        "$UNSIGNED_DIR"/detached-sig-apply.sh "$UNSIGNED_DIR"/dist/Bitcoin-Qt.app "$SIGNED_APP_DIST_DIR"
```

Comment:
Since you are invoking `osslsigncode` directly for the windows, perhaps the same should be done here with `signapple`? The apply script doesn't really do much anymore as `signapple` does a lot internally.